diff --git a/src/bitcoind.cpp b/src/bitcoind.cpp
index 543eba0..a10aa33 100644
--- a/src/bitcoind.cpp
+++ b/src/bitcoind.cpp
@@ -139,11 +139,11 @@ bool AppInit(int argc, char* argv[])
             // InitError will have been called with detailed error, which ends up on console
             exit(EXIT_FAILURE);
         }
-        if (!AppInitSanityChecks())
-        {
-            // InitError will have been called with detailed error, which ends up on console
-            exit(EXIT_FAILURE);
-        }
+        // if (!AppInitSanityChecks())
+        // {
+        //     // InitError will have been called with detailed error, which ends up on console
+        //     exit(EXIT_FAILURE);
+        // }
         if (gArgs.GetBoolArg("-daemon", false))
         {
 #if HAVE_DECL_DAEMON
