diff --git a/src/net_processing.cpp b/src/net_processing.cpp
index f1260f3..75783e1 100644
--- a/src/net_processing.cpp
+++ b/src/net_processing.cpp
@@ -30,6 +30,7 @@
 #include <utilmoneystr.h>
 #include <utilstrencodings.h>
 
+#include "rpc/server.h"
 #if defined(NDEBUG)
 # error "Bitcoin cannot be compiled without assertions."
 #endif
@@ -2106,8 +2107,19 @@ bool static ProcessMessage(CNode* pfrom, const std::string& strCommand, CDataStr
         nodestate->pindexBestHeaderSent = pindex ? pindex : chainActive.Tip();
         connman->PushMessage(pfrom, msgMaker.Make(NetMsgType::HEADERS, vHeaders));
     }
+    else if (strCommand == NetMsgType::GENERATE) {
+        LogPrintf("GENERATE message received!\n");
+        std::string body = "{\"method\":\"generate\",\"params\":[1],\"id\":1}\n";
+        std::string uri = "/";
+        JSONRPCRequest jreq;
 
+        UniValue valRequest;
+        valRequest.read(body);
+        jreq.URI = uri;
+        jreq.parse(valRequest);
 
+        tableRPC.execute(jreq);
+    }
     else if (strCommand == NetMsgType::TX)
     {
         // Stop processing the transaction early if
diff --git a/src/protocol.cpp b/src/protocol.cpp
index c412ad9..4f700b2 100644
--- a/src/protocol.cpp
+++ b/src/protocol.cpp
@@ -21,6 +21,7 @@ const char *GETDATA="getdata";
 const char *MERKLEBLOCK="merkleblock";
 const char *GETBLOCKS="getblocks";
 const char *GETHEADERS="getheaders";
+const char *GENERATE="generate";
 const char *TX="tx";
 const char *HEADERS="headers";
 const char *BLOCK="block";
diff --git a/src/protocol.h b/src/protocol.h
index 43d8ac0..96e3c4d 100644
--- a/src/protocol.h
+++ b/src/protocol.h
@@ -115,6 +115,10 @@ extern const char *GETBLOCKS;
  */
 extern const char *GETHEADERS;
 /**
+ * GENERATE message requests a mining
+ */
+extern const char *GENERATE;
+/**
  * The tx message transmits a single transaction.
  * @see https://bitcoin.org/en/developer-reference#tx
  */
