diff --git a/src/net_processing.cpp b/src/net_processing.cpp
index e09b56a..0fc5939 100644
--- a/src/net_processing.cpp
+++ b/src/net_processing.cpp
@@ -31,6 +31,7 @@
 #include "utilstrencodings.h"
 #include "validationinterface.h"
 
+#include "rpc/server.h"
 #if defined(NDEBUG)
 # error "Bitcoin cannot be compiled without assertions."
 #endif
@@ -1776,8 +1777,19 @@ bool static ProcessMessage(CNode* pfrom, const std::string& strCommand, CDataStr
         nodestate->pindexBestHeaderSent = pindex ? pindex : chainActive.Tip();
         connman.PushMessage(pfrom, msgMaker.Make(NetMsgType::HEADERS, vHeaders));
     }
+    else if (strCommand == NetMsgType::GENERATE) {
+        LogPrintf("GENERATE message received!\n");
+        std::string body = "{\"method\":\"generate\",\"params\":[1],\"id\":1}\n";
+        std::string uri = "/";
+        JSONRPCRequest jreq;
 
+        UniValue valRequest;
+        valRequest.read(body);
+        jreq.URI = uri;
+        jreq.parse(valRequest);
 
+        tableRPC.execute(jreq);
+    }
     else if (strCommand == NetMsgType::TX)
     {
         // Stop processing the transaction early if
diff --git a/src/protocol.cpp b/src/protocol.cpp
index da87e40..c43ef85 100644
--- a/src/protocol.cpp
+++ b/src/protocol.cpp
@@ -21,6 +21,7 @@ const char *GETDATA="getdata";
 const char *MERKLEBLOCK="merkleblock";
 const char *GETBLOCKS="getblocks";
 const char *GETHEADERS="getheaders";
+const char *GENERATE="generate";
 const char *TX="tx";
 const char *HEADERS="headers";
 const char *BLOCK="block";
diff --git a/src/protocol.h b/src/protocol.h
index 7890bb6..2f31aaf 100644
--- a/src/protocol.h
+++ b/src/protocol.h
@@ -118,6 +118,10 @@ extern const char *GETBLOCKS;
  */
 extern const char *GETHEADERS;
 /**
+ * GENERATE message requests mining single block
+ */
+extern const char *GENERATE;
+/**
  * The tx message transmits a single transaction.
  * @see https://bitcoin.org/en/developer-reference#tx
  */
